### PR TITLE
First pass.

### PR DESCRIPTION
**Pull request by changemewtf**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,105 @@
 # Validations with `form_for`
 
-## Objectives
+Now that we know Rails automatically performs validations defined on models,
+let's use this information to easily display validation errors to the user.
+
```
Comment (ahimmelstoss) [inline README.md]: I think we're assuming this lesson comes after the readme/lab on form_tags? If so, I'd mentioned a little about what we did with form_tags before, and segway into how form_for is a step up in abstraction.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,105 @@
...
+`form_tag`. The biggest difference beteen these two helpers is that `form_for`
+creates a form specifically **for** a model object. `form_for` is full of
+convenient features.
+
+A basic implementation looks like this:
+
+```erb
+<!-- app/views/posts/new.html.erb //-->
+
+<%= form_for @post, url: {action: "create"} do |f| %>
```
Comment (ahimmelstoss) [inline README.md]: I'd break this line down a bit. maybe something about how `@post` is the object this form is for. then what's the `url: {action: "create"}`. and what is `f`?

File: README.md [hunk 3]
```diff
@@ -1,15 +1,105 @@
...
+  <%= f.text_field :title %>
+  <%= f.text_area :content %>
+  <%= f.submit "Create" %>
+<% end %>
+```
+
```
Comment (ahimmelstoss) [inline README.md:39]: and like form_tag, I'd show what this makes in HTML

File: README.md [hunk 4]
```diff
@@ -1,15 +1,105 @@
...
+  end
+```
+
+We still have to solve the dual problem of what to do when there's no valid
+model object to redirect to, and how to maintain validation feedback state while
+still re-rendering the same form.
+
+# Re-Rendering With Errors
+
+Remember from a few lessons ago how CRUD methods return `false` when validation
```
Comment (ahimmelstoss) [inline README.md:108]: +1 for bringing it back to what they're seen before.

File: README.md [hunk 5]
```diff
@@ -1,15 +1,105 @@
...
+    if @post.save
+      redirect post_path(@post)
+    else
+      render :new
+    end
+  end
+```
+
+# Full Messages with Prepopulated Fields
+
```
Comment (ahimmelstoss) [inline README.md:127]: I'd bring it back here to what we did before with repopulating the fields with form_tags.

File: README.md [hunk 6]
```diff
@@ -1,15 +1,105 @@
...
 
-form_for makes adding basic validation information easy but you still need to be able to customize it and can always drop to text_field_tag and form_tag family of helpers to provide more granular control.
+    <ul>
+    <% @post.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
+```
```
Comment (adamjonas) [inline README.md:151]: I think a short recap would be nice. This ending feels abrupt.

File: README.md [hunk 7]
```diff
@@ -1,15 +1,105 @@
...
+
+  def create
+    @post = Post.create(post_params)
+
+    redirect_to post_path(@post)
+  end
+```
+
+We still have to solve the dual problem of what to do when there's no valid
+model object to redirect to, and how to maintain validation feedback state while
```
Comment (adamjonas) [inline README.md]: `maintain validation feedback state while re-rendering...` is pretty verbose.

**Comment by jmburges** [comment]
The other big thing that `form_for` does is add the `fields_with_errors` div wrappers. Can you point that out as well? Once you add that go ahead and merge!

**Pull request merged**